hs_core/models.py

```diff
@@ -589,6 +608,13 @@ def update(cls, element_id, **kwargs):
             kwargs['identifiers'] = identifiers
 
         party = super(Party, cls).update(element_id, **kwargs)
+        if set_active_user_flag:
```

Comment:
Could we refactor so that we aren't checking `party.hydroshare_user_id is...` a bunch of times? In other words, either move this logic up above (into the conditional block that starts ~line 595) or move that logic down to ~line 611.

Reply:
Good suggestion. Have cleaned it up now so the code is bit simpler.